crates/loader/src/local/assets.rs

```diff
@@ -20,22 +20,49 @@ use super::config::{RawDirectoryPlacement, RawFileMount};
 pub(crate) async fn prepare_component(
     raw_mounts: &[RawFileMount],
     src: impl AsRef<Path>,
-    base_dst: impl AsRef<Path>,
+    base_dst: Option<impl AsRef<Path>>,
     id: &str,
     exclude_files: &[String],
 ) -> Result<Vec<DirectoryMount>> {
-    tracing::info!(
-        "Mounting files from '{}' to '{}'",
-        src.as_ref().display(),
-        base_dst.as_ref().display()
-    );
-
-    let files = collect(raw_mounts, exclude_files, src)?;
-    let host = create_dir(&base_dst, id).await?;
-    let guest = "/".to_string();
-    copy_all(&files, &host).await?;
+    if let Some(base_dst) = base_dst {
+        tracing::info!(
+            "Mounting files from '{}' to '{}'",
+            src.as_ref().display(),
+            base_dst.as_ref().display()
+        );
 
-    Ok(vec![DirectoryMount { guest, host }])
+        let files = collect(raw_mounts, exclude_files, src)?;
+        let host = create_dir(&base_dst, id).await?;
+        let guest = "/".to_string();
+        copy_all(&files, &host).await?;
+
+        Ok(vec![DirectoryMount { guest, host }])
+    } else {
+        tracing::info!("directly mounting local asset directories into guest");
```

Comment:
I think the if and else parts of this function are entirely distinct.  As both are quite long and detailed, I'd be keen to split them into two separate functions (with this function just dispatching to one or the other) so that the reader has clearer breaks and less indentation!